### PR TITLE
[DOC] More changes to inheritance.

### PR DESCRIPTION
**Pull request by chriseth**
References #7970

### CODE REVIEW

File: docs/contracts/inheritance.rst [hunk 1]
```diff
@@ -51,6 +51,8 @@ Details are given in the following example.
     // contracts can access all non-private members including
     // internal functions and state variables. These cannot be
     // accessed externally via `this`, though.
+    // The keyword `virtual` means that the function can change
+    // its behaviour in derived classes ("overriding").
     contract Mortal is Owned {
```
Comment (bshastry) [inline docs/contracts/inheritance.rst:54]: The newly added comments (2 lines) can be moved here right before function definition

File: docs/contracts/inheritance.rst [hunk 2]
```diff
@@ -477,6 +490,10 @@ The reason for this is that ``C`` requests ``X`` to override ``A``
 requests to override ``X``, which is a contradiction that
 cannot be resolved.
 
+Due to the fact that you have to explicitly override a function
+that is inherited from multiple bases without a unique override,
+C3 linearization is not too important in practice.
```
Comment (bshastry) [inline docs/contracts/inheritance.rst:495]: Can `C3 linearization` be briefly explained?

File: docs/contracts/inheritance.rst [hunk 3]
```diff
@@ -521,4 +538,4 @@ Inheriting Different Kinds of Members of the Same Name
 
 When the inheritance results in a contract with a function and a modifier of the same name, it is considered as an error.
```
Comment (bshastry) [inline docs/contracts/inheritance.rst]:
```suggestion
We consider it an error when any of the following pairs in a contract have the same name due to inheritance
  - a function and a modifier
  - an event and a modifier
  - a function and an event
```

File: docs/contracts/inheritance.rst [hunk 4]
```diff
@@ -521,4 +538,4 @@ Inheriting Different Kinds of Members of the Same Name
...
 This error is produced also by an event and a modifier of the same name, and a function and an event of the same name.
```
Comment (bshastry) [inline docs/contracts/inheritance.rst]:
```suggestion

```

**Review by bshastry — COMMENTED** [review]
Added some minor comments for clarity

**Comment by leonardoalt** [comment]
Rebased

File: docs/contracts/inheritance.rst (hunk 2 above)
Comment (chriseth) [reply docs/contracts/inheritance.rst:495]: C3 is introduced in this section which contains a link to a wikipedia article about it. It cannot be briefly explained, but its properties are briefly explained above.

**Comment by chriseth** [comment]
Updated.

**Review by bshastry — APPROVED** [review]

**Pull request merged**